src/ec/eddsa.rs

```diff
@@ -18,9 +18,76 @@
 
 use {bssl, c, signature, signature_impl};
 use input::Input;
+use rand::{RAND, SecureRandom};
 
 struct EdDSA;
 
+/// An Ed25519 key pair.
+//
+// This can be used for signing. Allows generation of a new key pair through
+// the `generate()` method, or using an existing key pair with `from_bytes()`.
+pub struct Ed25519KeyPair {
+    private_public: [u8; 64],
+}
+
+
+impl<'a> Ed25519KeyPair {
+    /// Generate a new key pair from randomness.
+    pub fn generate(rng: &SecureRandom) -> Ed25519KeyPair {
+        let mut rand = RAND::new(rng);
+        let mut pair = Ed25519KeyPair { private_public: [0; 64] };
+        let mut public_key = [0u8; 32];
+        unsafe {
+            ED25519_keypair(public_key.as_mut_ptr(),
```

Comment:
It would be great if we remove the `out_public_key` argument of `ED25519_keypair` completely, in a follow-up commit.